File: projectBlueWater/src/main/java/com/lasthopesoftware/bluewater/client/library/items/media/files/properties/repository/FilePropertiesContainer.java
package com.lasthopesoftware.bluewater.client.library.items.media.files.properties.repository;

import java.util.Map;
import java.util.concurrent.ConcurrentHashMap;

public class FilePropertiesContainer {
	public final int revision;
	private final ConcurrentHashMap<String, String> properties;

	public FilePropertiesContainer(Integer revision, Map<String, String> properties) {
		this.revision = revision;
		this.properties = new ConcurrentHashMap<>(properties);
	}

	public Map<String, String> getProperties() {
		return properties;
	}

	public void updateProperty(String key, String value) {
		properties.put(key, value);
	}
}

File: projectBlueWater/src/main/java/com/lasthopesoftware/bluewater/client/browsing/items/media/files/properties/repository/FilePropertiesContainer.kt
package com.lasthopesoftware.bluewater.client.browsing.items.media.files.properties.repository

import java.util.concurrent.ConcurrentHashMap

class FilePropertiesContainer(val revision: Int, properties: Map<String, String>) {
	val properties: ConcurrentHashMap<String, String> = ConcurrentHashMap(properties)

	fun updateProperty(key: String, value: String) {
		properties[key] = value
	}
}
